### DISCUSSION TITLE
How to add piwheels into poetry.lock and exported requirements.txt ?!?

### DISCUSSION THREAD

**Discussion by jedie** [python-poetry/poetry#4816] [General]
I try to fix the installation of matplotlib on a Raspberry Pi, see: [install matplotlib binary doesn't work in RPi! django-for-runners_ynh/issues/19](https://github.com/YunoHost-Apps/django-for-runners_ynh/issues/19)

The problem: Only [piwheels.org](https://www.piwheels.org/project/matplotlib/) contains the arm wheel binary and [pypi.org](https://pypi.org/project/matplotlib/) not. I create the [poetry.lock](https://github.com/YunoHost-Apps/django-for-runners_ynh/blob/testing/poetry.lock) on my non ARM system, so it doesn't contain the arm wheel.

The result is, that the installation taks hours on RPi, because it compiles matplotlib from sources :(

I tried to add this to my pyproject.toml:
```diff
+[[tool.poetry.source]]
+name = "piwheels.org"
+url = "https://www.piwheels.org/simple"
+secondary = true
```

But this also doesn't insert the piwheels arm wheels in [poetry.lock](https://github.com/YunoHost-Apps/django-for-runners_ynh/blob/testing/poetry.lock) and/or exported [requirements.txt](https://github.com/YunoHost-Apps/django-for-runners_ynh/blob/testing/conf/requirements.txt) :(

Any idea?

**Comment by MichaIng** [comment]
See here: https://github.com/python-poetry/poetry/issues/8536#issuecomment-3403455948
Generally, Poetry seems to query only a single source for each package, no matter how priorities are set. It can use supplementary sources for packages which are not available on the primary source at all, but as fast as the primary source contains a single version or just the tarball for a package, supplemental sources are never considered.
